hack/update-json-library.sh

```diff
@@ -0,0 +1,27 @@
+# This script can be called via ./hack/update-json-library.sh` to update the
+# go-json-experiment fork in the repo. 
+
+GO_JSON_EXPERIMENT_DIR="pkg/internal/third_party/go-json-experiment/json"
+rm -rf $GO_JSON_EXPERIMENT_DIR
+git clone --depth 1 https://github.com/go-json-experiment/json $GO_JSON_EXPERIMENT_DIR
+cd $GO_JSON_EXPERIMENT_DIR
+HASH=$(git rev-parse HEAD)
```

Comment:
would it make more sense to set the hash we consume in this script, rather than always taking HEAD?

that would also mean that a presubmit could just run the script and check that there's no diff to verify

Reply:
I don't have a sense for whether the go-json-experiment repo always treats HEAD as stable / releasable, or if we might be grabbing a series of commits at an unfortunate point in time.

Specifying the hash we grab seems good for that reason, and I'd recommend getting an ack from dsnet when we're thinking of bumping.

Reply:
Added the hash as a variable in the script. Definitely agree we should get an ack from dsnet on version bumps, I'm also imagining this script will be temporary until `go-json-experiment` becomes stable.

Reply:
Nothing is so permanent as a temporary solution :-D

Reply:
Should we ... make the hash a parameter of the script so that we can say, use "HEAD" in a CI script to test the latest version and see if it still works or experiment?

Reply:
overrideable default could work ... `HASH="${HASH:-540f014424240312547dcccddf11a8229ca0f463}"` or something

Reply:
Yeah, agreed, though I don't know if I want to change the script everytime we change the hash, whatever.
Also could be ...
```bash
HASH="${HASH:-$(cat JSON-EXPERIMENTAL-HASH)}"
```

Reply:
Hmm, I would prefer that the hash is hard coded since we won't ever need to figure out which revision the repo is on. It's a bit of an extra step, but we can easily get the latest version by finding the latest commit hash and replacing it with that.

Reply:
Adopted `HASH="${HASH:-$(cat ./hack/JSON-EXPERIMENTAL-HASH)}"`. A bit unsure if this file is the best way to express this variable, but it's been updated.